pkg/cmd/run.go

```diff
@@ -111,41 +121,89 @@ doppler run --command "YOUR_COMMAND && YOUR_OTHER_COMMAND"`,
+		fallbackOpts := fallbackOptions{
+			enable:             enableFallback,
+			path:               fallbackPath,
+			legacyPath:         legacyFallbackPath,
+			readonly:           fallbackReadonly,
+			exclusive:          fallbackOnly,
+			exitOnWriteFailure: exitOnWriteFailure,
+			passphrase:         passphrase,
+		}
+		// retrieve secrets
+		dopplerSecrets := fetchSecrets(localConfig, enableCache, fallbackOpts, metadataPath, nameTransformer)
 
 		if preserveEnv {
 			utils.LogWarning("Ignoring Doppler secrets already defined in the environment due to --preserve-env flag")
 		}
 
-		env := os.Environ()
+		mountPath := cmd.Flag("mount").Value.String()
+		mountFormat := cmd.Flag("mount-format").Value.String()
+		maxReads := utils.GetIntFlag(cmd, "mount-max-reads", 32)
+		// only auto-detect the format if it hasn't been explicitly specified
+		shouldAutoDetectFormat := !cmd.Flags().Changed("mount-format")
+
+		originalEnv := os.Environ()
 		existingEnvKeys := map[string]bool{}
-		for _, envVar := range env {
+		for _, envVar := range originalEnv {
 			// key=value format
 			parts := strings.SplitN(envVar, "=", 2)
 			key := parts[0]
 			existingEnvKeys[key] = true
 		}
 
+		secrets := map[string]string{}
 		excludedKeys := []string{"PATH", "PS1", "HOME"}
-		for name, value := range secrets {
+		for name, value := range dopplerSecrets {
 			useSecret := true
 			for _, excludedKey := range excludedKeys {
 				if excludedKey == name {
+					utils.LogDebug(fmt.Sprintf("Ignoring restricted secret %s", name))
 					useSecret = false
 					break
 				}
 			}
+			if !useSecret {
+				continue
+			}
 
-			if useSecret && preserveEnv {
+			if preserveEnv && existingEnvKeys[name] == true {
 				// skip secret if environment already contains variable w/ same name
-				if existingEnvKeys[name] == true {
-					utils.LogDebug(fmt.Sprintf("Ignoring Doppler secret %s", name))
-					useSecret = false
+				utils.LogDebug(fmt.Sprintf("Ignoring Doppler secret %s", name))
+				continue
+			}
+
+			secrets[name] = value
+		}
+
+		env := originalEnv
+
+		var onExit func()
+		shouldMountFile := mountPath != ""
+		if shouldMountFile {
+			if shouldAutoDetectFormat {
+				if strings.HasSuffix(mountPath, ".env") {
+					mountFormat = "env"
+					utils.LogDebug(fmt.Sprintf("Detected %s format", mountFormat))
+				} else if strings.HasSuffix(mountPath, ".json") {
+					mountFormat = "json"
+					utils.LogDebug(fmt.Sprintf("Detected %s format", mountFormat))
 				}
 			}
 
-			if useSecret {
-				env = append(env, fmt.Sprintf("%s=%s", name, value))
+			absMountPath, handler, err := controllers.MountSecrets(secrets, mountFormat, mountPath, maxReads)
+			if !err.IsNil() {
+				utils.HandleError(err.Unwrap(), err.Message)
+			}
+			mountPath = absMountPath
+			onExit = handler
+
+			// export path to mounted file
+			env = append(env, fmt.Sprintf("%s=%s", "DOPPLER_CLI_SECRETS_PATH", mountPath))
```

Comment:
Suggestion to go with "mount" over "secrets" and condense the var name.

```suggestion
			env = append(env, fmt.Sprintf("%s=%s", "DOPPLER_MOUNT_PATH", mountPath))
```

Reply:
I like the change from `SECRETS` -> `MOUNT`. I'm leaning towards keeping the `CLI` prefix so that this variable is namespaced properly. There's currently nothing stopping the user from having a `DOPPLER_MOUNT_PATH` secret in their config, but we do prevent users from creating `DOPPLER_CLI` secrets.

Reply:
Ok got it. Whatever you think is best.